canal/canal_test.go

```diff
@@ -138,6 +138,23 @@ func (s *canalTestSuite) TestCanal() {
 	require.NoError(s.T(), err)
 }
 
+func (s *canalTestSuite) TestAnalyzeAdvancesSyncedPos() {
+	<-s.c.WaitDumpDone()
+
+	// We should not need to use FLUSH BINARY LOGS
+	// An ANALYZE TABLE statement should advance the saved position.
+	// There are still cases that don't advance, such as
+	// statements that won't parse like [CREATE|DROP] TRIGGER.
+	s.c.cfg.DisableFlushBinlogWhileWaiting = true
```

Comment:
The database user in this test has enough privilege for FLUSH BINARY LOGS, so it will not ensure FLUSH BINARY LOGS is skipped. I think we can get `s.c.GetMasterPos()` before and after `s.execute("ANALYZE TABLE test.canal_test")` and compare the binlog filename is not changed.